### DISCUSSION TITLE
Light/dark switch

### DISCUSSION THREAD

**Discussion by b00f** [vuepress-theme-hope/vuepress-theme-hope#585] [Q&A] [answered]
Is there any way to have simple dark/light switch like [docusaurus](https://v2.docusaurus.io/)?

**Comment by Mister-Hope** [comment, marked as answer]
https://vuepress-theme-hope.github.io/guide/interface/darkmode/

Please use `'switch'` option.

**Comment by b00f** [comment]
This is no what I am looking. Your answer implies that users have to choose only what Hope-theme offers. 
I think it' good to give more choices for user, like:
- Simple dark/light switch (only switch)
- No fantasy graphic motions

**Reply by Mister-Hope** [reply]
> I think it' good to give more choices for user, like:
> - Simple dark/light switch (only switch)
> - No fantasy graphic motions


It's good. Like I said in your other question, I have plans to build a pure mode.

> This is no what I am looking. Your answer implies that users have to choose only what Hope-theme offers. 

But, either you use my theme and choose what I provide, either you use other theme or other doc tools besides vuepress.l, either you extend my theme and overide `<ThemeOption />`. It's up to you to choose 2 and 3. You do have to only when you choose the first one.

I am not supposed to and not going to satisfy everyone.

**Reply by b00f** [reply]
Unfortunately I couldn't extend Hope theme and override <ThemeOption />. 

I believe Hope should provide more options for users.

**Reply by Mister-Hope** [reply]
> Unfortunately I couldn't extend Hope theme and override .
> 
> I believe Hope should provide more options for users.

Why can't you extend? vuepress-theme-hope is not extending any theme,. you can extends it through https://v1.vuepress.vuejs.org/theme/inheritance.html

Also I am providing a theme-eject function, see https://vuepress-theme-hope.github.io/guide/eject/.

I am already offering the best I can provide, bro.

**Comment by b00f** [comment]
Any progress on pure feature?

**Reply by Mister-Hope** [reply]
Pure feature will be included in V2.

Since I am using `balloon-css` in V1, and it can be hard to dynamicly add and remove attributes when using the template, I gave up importing this feature in V1. I may change thousands of lines in  dozens of files. And that's a bit complicated.

V2 will miragate to sass from stylus, and a lot of components full of state will change to render functions from template, that's a good time including this feature.

**Reply by b00f** [reply]
Thanks for your hard working. This is amazing that you have a lots of improvement in the next version. I wish I could help you, but I am afraid that I am not very familiar with VueJS.
